Assets/Scripts/Framework/Gameplay/LevelTracker.cs

```diff
@@ -0,0 +1,161 @@
+                    availableSpawns.Shuffle();
+                    Transform targetSpawn = availableSpawns[0];
+
+                    if (usedSpawns.Contains(targetSpawn))
+                        continue;
+
+                    GameObject newPackage = Instantiate(_packagePrefab);
+                    BoxGrabbable boxGrabbable = newPackage.GetComponent<BoxGrabbable>();
+
+                    ServerManager.Spawn(newPackage);
+
+                    newPackage.transform.position = targetSpawn.position;
+                    newPackage.transform.rotation = targetSpawn.rotation;
+
+                    amountToSpawn--;
+                    usedSpawns.Add(targetSpawn);
+                    availableSpawns.Remove(targetSpawn);
+
+                    _totalScore += _succesfullDeliveryBonus + boxGrabbable.GetComponent<BoxDamageable>().Health;
+
+                    boxGrabbable.UpdateLabelAndID(labels[i], i);
+                    boxGrabbable.PlaceOnGround();
+                }
+            }
+
+            List<Transform> freeSpawns = new();
+
+            for (int i = 0; i < _allStations.Length; i++)
+                for (int j = 0; j < _allStations[i].SpawnTransforms.Length; j++)
+                    if (!usedSpawns.Contains(_allStations[i].SpawnTransforms[j]))
+                        freeSpawns.Add(_allStations[i].SpawnTransforms[j]);
+
+            int fakeDeliverySpawns = random.Next(0, freeSpawns.Count);
+            freeSpawns.Shuffle();
+
+            for (int i = 0; i < fakeDeliverySpawns; i++)
+            {
+                string label = "";
+                while (label == "" || labels.Contains(label))
+                {
+                    label
+                        = CHARACTERS[random.Next(0, CHARACTERS.Length)].ToString()
+                        + CHARACTERS[random.Next(0, CHARACTERS.Length)].ToString();
+                }
+
+                GameObject newFakeDelivery = Instantiate(_packagePrefab);
+                BoxGrabbable boxGrabbable = newFakeDelivery.GetComponent<BoxGrabbable>();
+                ServerManager.Spawn(newFakeDelivery);
+
+                newFakeDelivery.transform.position = freeSpawns[i].position;
+                newFakeDelivery.transform.rotation = freeSpawns[i].rotation;
+
+                boxGrabbable.PlaceOnGround();
+                boxGrabbable.UpdateLabelAndID(label, -1);
+            }
+        }
+
+        /// <summary>
+        /// A function that handles the delivery of a package.
+        /// </summary>
+        /// <param name="delivery">The package that was delivered.</param>
+        /// <param name="stationID">The ID of the station it was delivered to.</param>
+        public void HandlePackageDelivery(BoxGrabbable delivery, int stationID)
+        {
+            if (delivery.PackageID == stationID)
+                _currentScore += _succesfullDeliveryBonus + delivery.GetComponent<BoxDamageable>().Health;
+            else
+                _currentScore -= _succesfullDeliveryBonus;
```

Comment:
Maybe would be nice if we subtract  a separate value from `_currentScore`  if unsuccessful for balancing tweaks.